pandas/io/formats/style.py

```diff
@@ -499,6 +501,49 @@ def format(self, formatter, subset=None, na_rep: Optional[str] = None) -> "Style
                 self._display_funcs[(i, j)] = formatter
         return self
 
+    def set_data_classes(self, classes: DataFrame) -> "Styler":
+        """
+        Add string based CSS class names to data cells that will appear in the
+        `Styler` HTML result.
+
+        Parameters
+        ----------
+        classes : DataFrame
+            DataFrame containing strings that will be translated to CSS classes. Empty
+            strings, None, or NaN values will be ignored. DataFrame must have
+            identical rows and columns to the underlying `Styler` data.
```

Comment:
I think it'd be reasonable to reindex `classes` with the index / columns of `self`. Then you don't need to fill with empty strings for the rows /cols you don't care about.

Reply:
yes OK good idea, I'll take a look at implementing this..